jobsdb/jobsdb.go

```diff
@@ -118,43 +118,149 @@ var getTimeNowFunc = func() time.Time {
 	return time.Now()
 }
 
+// AppendSafeTx sealed interface
+type AppendSafeTx interface {
+	Tx() *sql.Tx
+	appendSafeTxIdentifier() string
+}
+
+type appendSafeTx struct {
+	tx       *sql.Tx
+	identity string
+}
+
+func (r *appendSafeTx) appendSafeTxIdentifier() string {
+	return r.identity
+}
+
+func (r *appendSafeTx) Tx() *sql.Tx {
+	return r.tx
+}
+
+// EmptyAppendSafeTx returns an empty interface usable only for tests
+func EmptyAppendSafeTx() AppendSafeTx {
+	return &appendSafeTx{}
```

Comment:
why we need that? I thought we can just mock the AppendSafeTx public interface in the tests, cant' we?

Reply:
Unfortunately we cannot mock this since it is a sealed interface